### ISSUE TITLE
[lecture10[0619] Batch Normalization

### ISSUE THREAD

**Issue by jwemilee** [PiLab-CAU/ComputerVision-2401#62]
Hello,

I have a question regarding batch normalization(lec 10_2, 33p): 
<img width="948" alt="batch" src="https://github.com/PiLab-CAU/ComputerVision-2401/assets/80567485/aa79092e-aedf-425b-996b-cf418d1bb3f0">

The slide mentions that in some cases, we cannot increase the batch size due to huge computational requirements. How does batch normalization perform with very small batch sizes (e.g., 1 or 2)? Can other normalization techniques, such as group normalization, instance normalization, or layer normalization, mitigate any negative impacts in such scenarios?

Thank you for your time.

**Comment by yjyoo3312** [comment]
@jwemilee Thank you for the comment!

As you mentioned, there are scenarios in training generative models where we cannot use a batch size larger than 1 or 2. In such cases, we consider instance normalization, layer normalization, or group normalization, as they do not calculate the batch-wise mean, so the batch size does not affect performance. 

However, if we have sufficient GPU memory, it is desirable to increase the batch size, and using batch normalization is a widely adopted approach. Of course, there are exceptions: for instance, transformers use layer normalization, and the use of group normalization with large-scale ResNet architectures has shown promising results compared to batch normalization.
